lms/djangoapps/mobile_api/users/views.py

```diff
@@ -258,35 +260,41 @@ class UserCourseEnrollmentsList(generics.ListAPIView):
           certified).
         * url: URL to the downloadable version of the certificate, if exists.
     """
-    queryset = CourseEnrollment.objects.all()
-    serializer_class = CourseEnrollmentSerializer
-    lookup_field = 'username'
-
-    # In Django Rest Framework v3, there is a default pagination
-    # class that transmutes the response data into a dictionary
-    # with pagination information.  The original response data (a list)
-    # is stored in a "results" value of the dictionary.
-    # For backwards compatibility with the existing API, we disable
-    # the default behavior by setting the pagination_class to None.
-    pagination_class = None
 
     def is_org(self, check_org, course_org):
         """
         Check course org matches request org param or no param provided
         """
         return check_org is None or (check_org.lower() == course_org.lower())
 
-    def get_queryset(self):
-        enrollments = self.queryset.filter(
-            user__username=self.kwargs['username'],
+    def get(self, request, username):
+        """
+        Returns a list of courses enrolled by user.
+        """
+        queryset = CourseEnrollment.objects.all()
+        course_ids = set(queryset.values_list('course_id', flat=True))
```

Comment:
This fetches all `CourseEnrollments` from the database (this query is completely untenable in production). You need to use `distinct` (https://docs.djangoproject.com/en/1.10/ref/models/querysets/#django.db.models.query.QuerySet.distinct)

cc @ormsbee